packages/eds-core-react/src/components/Datepicker/DatePicker.tsx

```diff
@@ -0,0 +1,211 @@
+import {
+  forwardRef,
+  RefObject,
+  useCallback,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from 'react'
+import { DatePickerProps } from './props'
+import { Calendar } from './calendars/Calendar'
+import { DateField } from './fields/DateField'
+import { calendar, warning_outlined } from '@equinor/eds-icons'
+import { useCommonHook } from './utils/useCommonHook'
+import { FieldWrapper } from './fields/FieldWrapper'
+import { Toggle } from './fields/Toggle'
+import {
+  CalendarDate,
+  CalendarDateTime,
+  createCalendar,
+  fromDate,
+  toCalendarDate,
+  toCalendarDateTime,
+} from '@internationalized/date'
+import { useDatePicker, useLocale } from 'react-aria'
+import { useDatePickerState } from '@react-stately/datepicker'
+import { DatePickerProvider, defaultTimezone } from './utils/context'
+import { HelperTextProps } from '../InputWrapper/HelperText'
+import { tokens } from '@equinor/eds-tokens'
+import { Icon } from '../Icon'
+
+/**
+ * DatePicker component encapsulates the logic for selecting a single date.
+ * Either by accessible input field or by a calendar.
+ */
+export const DatePicker = forwardRef<HTMLDivElement, DatePickerProps>(
+  (
+    {
+      onChange,
+      label,
+      value,
+      isDateUnavailable,
+      minValue,
+      maxValue,
+      footer,
+      Header,
+      timezone,
+      defaultValue,
+      time,
+      ...props
+    }: DatePickerProps,
+    ref,
+  ) => {
+    timezone = timezone ?? defaultTimezone
+    const [innerValue, setInnerValue] = useState<
+      CalendarDate | CalendarDateTime
+    >(
+      value
+        ? time
+          ? toCalendarDateTime(fromDate(value, timezone))
+          : toCalendarDate(fromDate(value, timezone))
+        : null,
+    )
+    const [open, setOpen] = useState<boolean | null>(null)
+    const inputRef = useRef<HTMLInputElement | null>(null)
+    const pickerRef = useRef<HTMLDivElement | null>(null)
+
+    const actualRef = useMemo(() => {
+      return (ref || inputRef) as RefObject<HTMLDivElement>
+    }, [ref, inputRef])
+
+    const { _minValue, _maxValue, _isDateUnavailable } = useCommonHook(
+      minValue,
+      maxValue,
+      isDateUnavailable,
+      timezone,
+    )
+
+    const _onChange = useCallback(
+      (value: CalendarDate | CalendarDateTime) => {
+        if (value) {
+          setInnerValue(
+            time ? toCalendarDateTime(value) : toCalendarDate(value),
+          )
+        } else {
+          setInnerValue(null)
+        }
+        if (open) {
+          setOpen(false)
+        }
+        if (onChange) {
+          if (!value) {
+            onChange(null)
+          } else {
+            const date = value.toDate(timezone)
+            onChange(date)
+          }
+        }
+      },
+      [onChange, open, time, timezone],
+    )
+
+    const _value = useMemo(() => {
```

Comment:
It's not super-expensive, but it does parse / convert the dates quite often without the useMemo.

@vnys / @oddvernes do you have a "standard" in the team for when to use / not use memo? It's highly likely that I overuse it as I hate doing more work / rerendering more often than is needed. I also think useMemo is more readable than just assigning a variable, but I'm also weird 🙈 so would be good with more opinions here.

Reply:
I can recommend this article about the subject. https://kentcdodds.com/blog/usememo-and-usecallback

Reply:
Yeah, I'm aware it's probably not super-necessary, but I tend to prefer using memo for stuff that causes cascading updates since it can be tough to measure the performance impact. However, I removed it for now, so it can be optimized at a later stage instead.